webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/NodeDetailsAPI.scala

```diff
@@ -0,0 +1,145 @@
+package com.normation.rudder.rest.internal
+
+import com.normation.inventory.domain.NodeId
+import com.normation.inventory.domain.Software
+import com.normation.inventory.services.core.ReadOnlySoftwareDAO
+import com.normation.rudder.domain.logger.TimingDebugLoggerPure
+import com.normation.rudder.domain.nodes.NodeInfo
+import com.normation.rudder.domain.policies.GlobalPolicyMode
+import com.normation.rudder.domain.policies.PolicyModeOverrides.Always
+import com.normation.rudder.domain.policies.PolicyModeOverrides.Unoverridable
+import com.normation.rudder.reports.execution.AgentRunWithNodeConfig
+import com.normation.rudder.reports.execution.RoReportsExecutionRepository
+import com.normation.rudder.repository.json.DataExtractor.OptionnalJson
+import com.normation.rudder.rest.RestExtractorService
+import com.normation.rudder.services.nodes.NodeInfoService
+import com.normation.rudder.web.components.DateFormaterService
+import com.typesafe.config.ConfigRenderOptions
+import net.liftweb.common.Box
+import net.liftweb.common.Full
+import net.liftweb.common.Loggable
+import net.liftweb.http.JsonResponse
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.Req
+import net.liftweb.http.rest.RestHelper
+import net.liftweb.json.JsonAST.JArray
+import net.liftweb.json.JsonAST.JField
+import net.liftweb.json.JsonAST.JNothing
+import net.liftweb.json.JsonAST.JObject
+import net.liftweb.json.JsonAST.JString
+import net.liftweb.json.parse
+import zio.ZIO
+
+class NodeDetailsAPI (
+  nodeInfoService: NodeInfoService
+  , restExtractor : RestExtractorService
+  , reportsExecutionRepository: RoReportsExecutionRepository
+  , getGlobalMode : () => Box[GlobalPolicyMode]
+  , readOnlySoftwareDAO: ReadOnlySoftwareDAO
+
+) extends  RestHelper with  Loggable {
+
+
+  def serialize(agentRunWithNodeConfig: Option[AgentRunWithNodeConfig], globalPolicyMode: GlobalPolicyMode, nodeInfo : NodeInfo, properties : List[String], softs: List[Software]) = {
+    import net.liftweb.json.JsonDSL._
+
+    val (policyMode,explanation) =
+      (globalPolicyMode.overridable,nodeInfo.policyMode) match {
+        case (Always,Some(mode)) =>
+          (mode,"<p>This mode is an override applied to this node. You can change it in the <i><b>node's settings</b></i>.</p>")
+        case (Always,None) =>
+          val expl = """<p>This mode is the globally defined default. You can change it in <i><b>settings</b></i>.</p><p>You can also override it on this node in the <i><b>node's settings</b></i>.</p>"""
+          (globalPolicyMode.mode, expl)
+        case (Unoverridable,_) =>
+          (globalPolicyMode.mode, "<p>This mode is the globally defined default. You can change it in <i><b>Settings</b></i>.</p>")
```

Comment:
I think it would be better if it were a css class: it offers more customizability, and prevent sending the same text thousands of time to the server

Reply:
Ok I think we can do it after feature freeze